code/modules/borer/borer.dm

```diff
@@ -0,0 +1,301 @@
+	var/datum/action/innate/borer/talk_to_borer/action_talk_to_borer = new
+	var/datum/action/innate/borer/talk_to_brain/action_talk_to_brain = new
+	var/datum/action/innate/borer/take_control/action_take_control = new
+	var/datum/action/innate/borer/give_back_control/action_give_back_control = new
+	var/datum/action/innate/borer/leave_body/action_leave_body = new
+	var/datum/action/innate/borer/make_chems/action_make_chems = new
+	var/datum/action/innate/borer/make_larvae/action_make_larvae = new
+	var/datum/action/innate/borer/freeze_victim/action_freeze_victim = new
+	var/datum/action/innate/borer/torment/action_torment = new
+	var/datum/action/innate/borer/scan_chems/action_scan_chems = new
+
+/mob/living/simple_animal/borer/proc/summon()
+	var/datum/emergency_call/custom/em_call = new()
+	em_call.name = "Cortical Borer"
+	em_call.mob_max = 1
+	em_call.players_to_offer = list(src)
+	em_call.owner = null
+
+	em_call.activate(announce = FALSE)
+
+	message_staff("A new Cortical Borer has spawned at [get_area(loc)]")
+
+/mob/living/simple_animal/borer/New(var/atom/newloc, var/gen=1, var/ERT = FALSE, var/reproduction = 0)
+	..(newloc)
+	generation = gen
+	add_language("Cortical Link")
+	real_name = "Cortical Borer [rand(1000,9999)]"
+	truename = "[borer_names[min(generation, borer_names.len)]] [rand(1000,9999)]"
+	can_reproduce = reproduction
+	GrantBorerActions()
+	GiveBorerHUD()
+	if((!is_admin_level(z)) && ERT)
+		summon()
+
+/mob/living/simple_animal/borer/proc/GiveBorerHUD()
+	var/datum/mob_hud/H = huds[MOB_HUD_MEDICAL_OBSERVER]
+	H.add_hud_to(src)
+
+/mob/living/simple_animal/borer/can_ventcrawl()
+	return TRUE
+
+/mob/living/simple_animal/borer/initialize_pass_flags(var/datum/pass_flags_container/PF)
+	..()
+	if (PF)
+		PF.flags_pass = PASS_MOB_THRU|PASS_FLAGS_CRAWLER
+		PF.flags_can_pass_all = PASS_ALL^PASS_OVER_THROW_ITEM
+
+/mob/living/simple_animal/borer/get_status_tab_items()
+	. = ..()
+
+	var/CR = "Yes"
+	if(!can_reproduce)
+		CR = "Forbidden"
+	else if((chemicals < BORER_LARVAE_COST))
+		CR = "No"
+
+	. += ""
+	. += "Borer:"
+	. += "Name: [truename]"
+	. += "Chemicals: [round(chemicals)]/[round(max_chems)]"
+	. += "Can Reproduce: [CR]"
+
+/mob/living/simple_animal/borer/say(message)
+	var/datum/language/lang = parse_language(message)
+	if(!istype(lang, /datum/language/corticalborer) && loc == host && !talk_inside_host)
+		to_chat(src, "<span class='warning'>You've disabled audible speech while inside a host! Re-enable it under the borer tab, or stick to borer communications.</span>")
+		return
+
+	. = ..()
+
+
+/mob/living/simple_animal/borer/Life(delta_time)
+	..()
+
+	if(host)
+		if(!stat && host.stat != DEAD)
+			if((host.reagents.has_reagent("sugar") && !host.reagents.has_reagent("benzyme")) || host.reagents.has_reagent("bcure"))
+				if(!docile)
+					if(controlling)
+						to_chat(host, SPAN_XENODANGER("You feel the soporific flow of sugar in your host's blood, lulling you into docility."))
```

Comment:
probs should also change this to 'a chemical in the blood makes you fall asleep' or whatnot, so it doesnt specify sugar

Reply:
They're meant to know what the prevention is, though I may look at changing it into a trait like you suggested.